### DISCUSSION TITLE
Realtime mode

### DISCUSSION THREAD

**Discussion by faxm0dem** [asciinema/discussions#193] [Ideas]
Wouldn't it be awesome to be able to share a session in realtime, as in streaming?

**Comment by ku1ik** [comment]
That would be hard to achieve with the current architecture but I am considering rethinking the system to make this possible, and maybe a core feature even.

**Comment by faxm0dem** [comment]
I'm convinced this could lead to a killer feature: every day I see ppl on IRC asking for shared screen or tmux sessions

**Comment by fuadsaud** [comment]
wow, that's huge.

**Comment by ku1ik** [comment]
This is happening :) With either 1.0 or 1.1 of asciinema recorder.

**Comment by faxm0dem** [comment]
Awesome!

**Comment by ku1ik** [comment]
@faxm0dem we're currently figuring out data format here: asciinema/asciinema-cli#70.

**Comment by taddeimania** [comment]
This could lead to some pretty amazing things in terms of broadcasting programming sessions.  I'm thinking educational / remote applications.  I'm very excited about this. :+1:

**Comment by lenka21** [comment]
what is the status of this task?

**Comment by ku1ik** [comment]
It's planned, I'm currently experimenting with several approaches to this.

**Comment by faxm0dem** [comment]
FWIW there's [gotty](/yudai/gotty) that does something similar and it's awesome

**Comment by xuefer** [comment]
i'm looking for how to share tty and find this, this is good. streaming is also needed
simply sharing tty is what i needed but recording feature like asciinema is also nice to have given someone may want to watch it later

**Comment by faxm0dem** [comment]
@xuefer another awesome alternative for streaming and recording is [tmux2html](/tweekmonster/tmux2html)

**Comment by bkil** [comment]
Related/duplicate: https://github.com/asciinema/discussions/issues/180
Related: https://github.com/asciinema/discussions/issues/4
Related: https://github.com/asciinema/asciinema/issues/516

**Comment by ku1ik** [comment]
FYI, you can now live stream your terminal with asciinema CLI 3.0 (https://github.com/asciinema/asciinema/releases/tag/v3.0.0-rc.4). Check out `asciinema stream --help` or `asciinema session --help`.
